src/metabase/sync/analyze/classify.clj

```diff
@@ -37,23 +39,29 @@
 
 (def ^:private values-that-can-be-set
   "Columns of Field that classifiers are allowed to set."
-  #{:special_type :preview_display})
+  #{:special_type :preview_display :entity_type})
 
-(s/defn ^:private ^:always-validate save-field-updates!
+(def ^:private FieldOrTableInstance (s/either i/FieldInstance i/TableInstance))
+
+(s/defn ^:private ^:always-validate save-model-updates!
   "Save the updates in UPDATED-FIELD."
-  [original-field :- i/FieldInstance, updated-field :- i/FieldInstance]
-  (let [[_ values-to-set] (data/diff original-field updated-field)]
+  [original-model :- FieldOrTableInstance, updated-model :- FieldOrTableInstance]
```

Comment:
Does this mean we'll only try to classify types if someone is analyzing the full database?

Reply:
Yes. But I was just following precedence and used the same spot where we do other similar detection. But with types becoming more important, we should probably move everything that does not touch the warehouse into the shallow sync.